template/.github/workflows/ci.yml

```diff
@@ -170,3 +170,59 @@ jobs:
 
     - name: Run the functional test suite
       run: poetry run invoke test --suite=functional
+
+build-and-test-docker-image:
+    name: Build and Test Docker Image
+    needs: [static-analysis-checks, type-checking, run-system-tests, run-functional-tests]
+    runs-on: ubuntu-latest
+
+    services:
+      postgres:
+        image: postgres:14
+        env:
+          POSTGRES_USER: dev
+          POSTGRES_PASSWORD: dev_password
+          POSTGRES_DB: dev
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+        ports:
+            # Allows us to use the same env config used locally by mapping to port to
+            # the test runner container.
+            - 5432:5432
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+
+    - name: Set up Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: '3.12'
+
+    - name: Install dependencies
+      run: |
+        pip install invoke
+        cp example.env .env
+
+    - name: Modify DATABASE_URL for GitHub Actions
+      run: |
+        echo 'DATABASE_URL=postgres://dev:dev_password@postgres:5432/dev'  >> .env
+        cat .env
+
+    - name: Build Docker Image
+      run: invoke build-image --tag abc
+
+    - name: Run Docker image
+      run: |
+        docker run --name test-container --env-file .env {{ project_name }}:abc
+        docker exec -it test-container django-admin check
```

Comment:
Please take a look at the `entrypoint.sh`. That file is the entry point of the docker container (see the final line of the Dockerfile). Note how it allows you can pass through values to trigger actions.

If no value is pass in it will run migrations and start the server. You do _not_ want to do this. Instead you want it to just run the `check` management command directly and then exit

`docker run --rm --env-file .env -it {{ project_name }}:abc check` 

Note the use of `--rm` to destroy the container after it exits.

Reply:
Thanks Jon for the clarification. You're right regarding the fact I need to pass the command here.
Re the check command, my understanding is that it's[ a django-admin command](https://docs.djangoproject.com/en/5.0/ref/django-admin/#check). Can we use it under `manage.py`?

Reply:
Good call! Yes I verified it locally.